From ce281f278bb56f4dca059fea34a97446989a9076 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.xyz>
Date: Thu, 28 Dec 2023 22:11:14 -0800
Subject: [PATCH 11/68] grub-mkrescue: specify -iso-level 3

This allows GRUB rescue ISOs to boot on LoongArch firmware.
---
 util/grub-mkrescue.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/util/grub-mkrescue.c b/util/grub-mkrescue.c
index 6dc71a8a1..dd1b3e921 100644
--- a/util/grub-mkrescue.c
+++ b/util/grub-mkrescue.c
@@ -518,6 +518,8 @@ main (int argc, char *argv[])
   xorriso_push ("-as");
   xorriso_push ("mkisofs");
   xorriso_push ("-graft-points");
+  xorriso_push ("-iso-level");
+  xorriso_push ("3");
 
   iso9660_dir = grub_util_make_temporary_dir ();
   grub_util_info ("temporary iso9660 dir is `%s'", iso9660_dir);
-- 
2.51.1

